pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -67,19 +67,27 @@ func NewEdgeFetcher(ctx context.Context, bucketName string, storeWatcher store.W
 		storeWatcher:     storeWatcher,
 		processorManager: manager,
 		lastProcessedWm:  lastProcessedWm,
-		log:              log,
+		log:              log.With("bucket", bucketName),
 	}
 }
 
-// GetWatermark gets the largest possible watermark from all the active processors of all partitions for the given offset and partition.
-// We calculate the watermark for the given offset and partition, update the lastProcessedWatermark of the given partition to the watermark we just calculate.
-// Then, we compare the lastProcessedWatermark from all partitions and return the minimum as the edge watermark.
-// deletes the processor if it's not active.
-func (e *edgeFetcher) GetWatermark(inputOffset isb.Offset, fromPartitionIdx int32) wmb.Watermark {
+// ProcessOffsetGetWatermark processes the offset on the partition indicated and returns the overall Watermark
+// from all Partitions
+func (e *edgeFetcher) ProcessOffsetGetWatermark(offset isb.Offset, fromPartitionIdx int32) wmb.Watermark {
```

Comment:
```suggestion
func (e *edgeFetcher) UpdateAndGetWatermark(offset isb.Offset, fromPartitionIdx int32) wmb.Watermark {
```

I think `ProcessOff` can be confusing since we use p